### PR TITLE
Generics support (Add support for List, Set, Map #11)

### PR DESCRIPTION
**Pull request by UltimaPhoenix**
Hello I’ve created this first pull request draft. I've added code only for hibernate 5.2 module and I’ve added a test only for one DBMS. If you think it is fine, I can improve the pull request by cloning the changes for the other hibernate modules and by adding tests for every DBMS.

[Related issue 11](https://github.com/vladmihalcea/hibernate-types/issues/11)

### CODE REVIEW

File: hibernate-types-52/src/test/java/com/vladmihalcea/hibernate/type/json/MySQLJsonTypeTest.java [hunk 1]
```diff
@@ -53,6 +56,7 @@ public void test() {
             Event event = new Event();
             event.setId(1L);
             event.setLocation(location);
+            event.setAlternativeLocations(Arrays.asList(location));
```
Comment (vladmihalcea) [inline hibernate-types-52/src/test/java/com/vladmihalcea/hibernate/type/json/MySQLJsonTypeTest.java]: I think it's better to have a separate Test class for this new feature.

**Comment by vladmihalcea** [comment]
Looks good. Just the single comment I added on the code review, with a separate test. Thanks.

File: hibernate-types-52/src/test/java/com/vladmihalcea/hibernate/type/json/MySQLJsonTypeTest.java (hunk 1 above)
Comment (UltimaPhoenix) [reply hibernate-types-52/src/test/java/com/vladmihalcea/hibernate/type/json/MySQLJsonTypeTest.java]: Ok

**Comment by vladmihalcea** [comment]
I tested it and it works nicely. Please update the PR for the other modules and I'll integrate it next week and release a new version.

**Comment by UltimaPhoenix** [comment]
Ok I'm working on it. I'm adding Travis CI too

File: .travis.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ before_install:
 before_script:
   - psql -c 'create database high_performance_java_persistence;' -U postgres
 script:
-  - mvn verify
+  - mvn verify -Ptravis
```
Comment (vladmihalcea) [inline .travis.yml]: Could you split this into two Pull Requests? I'd like to have the original one separated from the Travis one. You should create a separate issue for Travis.

Thanks.

Comment (UltimaPhoenix) [reply .travis.yml]: sure

**Comment by UltimaPhoenix** [comment]
I completed the pull request :blush:. You can safely review the pull request now.
I checked that in the hibernate 4 module there isn't any `json`/`jsonb` supported type :scream:, so I did nothing on that. :innocent:

**Comment by vladmihalcea** [comment]
Cool. I'll review it tomorrow.

**Comment by vladmihalcea** [comment]
Applied upstream, thanks

**Comment by vladmihalcea** [comment]
This feature is now available if you upgrade to 1.2.0.

**Pull request closed without merging**